smarts/protos/worker.proto

```diff
@@ -0,0 +1,52 @@
+// Copyright (C) 2020. Huawei Technologies Co., Ltd. All rights reserved.
+
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+syntax = "proto3";
+
+package worker;
+
+// Interface exported by the worker server.
+service Worker {
+
+    // Builds Agent according the AgentSpec.
+    rpc build(Specification) returns (Status) {}
+
+    // Agent processes observations and returns action.
+    rpc act(Observation) returns (Action) {}
+}
+
+// Agent specification
+message Specification {
+    bytes payload = 1;
+}
+
+// Status
+message Status {
```

Comment:
What can we return if the `Message` type is empty?

Reply:
gRPC mandates every remote procedure call to have exactly one input and one output, in line with good functional programming. We need to provide a `message`, albeit being empty, as it makes explicitly clear that this rpc does not return any value.

The current `message` also serves as a placeholder, whose structure can be updated to contain some return value in the future, if needed.